vyper/ast_utils.py

```diff
@@ -0,0 +1,141 @@
+import ast as python_ast
+
+import vyper.ast as vyper_ast
+from vyper.exceptions import (
+    ParserException,
+    SyntaxException,
+)
+from vyper.parser.parser_utils import (
+    annotate_and_optimize_ast,
+)
+from vyper.parser.pre_parser import (
+    pre_parse,
+)
+
+
+def parse_python_ast(source_code, node):
+    if isinstance(node, list):
+        o = []
```

Comment:
I'd advocate for moving towards a non-mutative generator based pattern for these types of accumulations.  I know it may not be familiar, but this approach has proven effective across a lot of code over the last three years and it *should* be more performant in most cases.

The smoothest way to do this would be vendoring a version of the `to_tuple` style decorators from the `eth-utils` library, but it can be done without those as well as long as you're careful with your generators or write wrappers.

```python
def real_thing():  # the wrapper
    return tuple(_actual_thing())

def _actual_thing():  # the actual functionality
    for i in range(100):
        yield i
```